chunk_arrow.go

```diff
@@ -67,3 +76,39 @@ func buildFirstArrowChunk(rowsetBase64 string) arrowResultChunk {
 
 	return arrowResultChunk{*rr, 0, 0, memory.NewGoAllocator()}
 }
+
+/**
+Writes []array.Record to array record channel. Returns number of rows from records written to channel
+*/
+func (arc *arrowResultChunk) writeToArrowChan(ch chan<- []array.Record) (int, error) {
+	var numRows int
+	var records []array.Record
+
+	for {
+		record, err := arc.reader.Read()
+		if err == io.EOF {
+			ch <- records
+			return numRows, nil
+		} else if err != nil {
```

Comment:
because you have an early return, I would also make this a new if statement

Reply:
this one is to keep it equal to their code above - they use `else if` with the same code